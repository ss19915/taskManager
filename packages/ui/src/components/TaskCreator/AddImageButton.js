import React from 'react';
import { Fab, AddAPhotoIcon } from '@task-manager/theme';
import T from 'prop-types';

const AddImageButton = (props) => {
    const { onChange } = props;
    const inputRef = React.createRef();

    const openFileChooser = () => { inputRef.current.click() };
    const getObjectValues = (object) => Object.values(object);
    return (
        <React.Fragment>
            <input
                hidden multiple
                ref={inputRef}
                type='file'
                onChange={({ target }) => onChange(getObjectValues(target.files))}
            />
            <Fab onClick={openFileChooser}>
                <AddAPhotoIcon />
            </Fab>
        </React.Fragment>
    );
};

AddImageButton.propTypes = {
    onChange: T.func.isRequired,
};

export default AddImageButton;