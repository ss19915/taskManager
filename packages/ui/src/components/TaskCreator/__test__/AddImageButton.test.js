import AddImageButton from '../AddImageButton';
import React from 'react';

describe('Component: AddImageButton', () => {
    it('Should match snapshot', () => {
        const wrapper = shallow(<AddImageButton onChange={() => { }} />);

        expect(wrapper).toMatchSnapshot();
    });
});