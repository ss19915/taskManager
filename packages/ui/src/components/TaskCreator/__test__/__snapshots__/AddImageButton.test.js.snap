// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: AddImageButton Should match snapshot 1`] = `
<Fragment>
  <input
    hidden={true}
    multiple={true}
    onChange={[Function]}
    type="file"
  />
  <WithStyles(ForwardRef(Fab))
    onClick={[Function]}
  >
    <AddAPhotoIcon />
  </WithStyles(ForwardRef(Fab))>
</Fragment>
`;
